lib/bulk/common.js

```diff
@@ -67,138 +73,140 @@ var LegacyOp = function(batchType, operation, index) {
  * @property {number} nRemoved Number of removed documents
  * @return {BulkWriteResult} a BulkWriteResult instance
  */
-var BulkWriteResult = function(bulkResult) {
-  defineReadOnlyProperty(this, 'ok', bulkResult.ok);
-  defineReadOnlyProperty(this, 'nInserted', bulkResult.nInserted);
-  defineReadOnlyProperty(this, 'nUpserted', bulkResult.nUpserted);
-  defineReadOnlyProperty(this, 'nMatched', bulkResult.nMatched);
-  defineReadOnlyProperty(this, 'nModified', bulkResult.nModified);
-  defineReadOnlyProperty(this, 'nRemoved', bulkResult.nRemoved);
+class BulkWriteResult {
+  constructor(bulkResult) {
+    defineReadOnlyProperty(this, 'ok', bulkResult.ok);
+    defineReadOnlyProperty(this, 'nInserted', bulkResult.nInserted);
+    defineReadOnlyProperty(this, 'nUpserted', bulkResult.nUpserted);
+    defineReadOnlyProperty(this, 'nMatched', bulkResult.nMatched);
+    defineReadOnlyProperty(this, 'nModified', bulkResult.nModified);
+    defineReadOnlyProperty(this, 'nRemoved', bulkResult.nRemoved);
+
+    /**
+     * Return an array of inserted ids
+     *
+     * @return {object[]}
+     */
+    this.getInsertedIds = function() {
```

Comment:
IMO if we're converting these to classes, then these should be defined as getters on the class. Since these properties close over the passed in `bulkResult`, it's already living for as long as the instance anyway - so may as well make it more readable by storing off the `bulkRest` and providing prototype methods for accessing this data.